packages/material-ui-lab/src/internal/pickers/PickersToolbarButton.tsx

```diff
@@ -1,59 +1,81 @@
 import * as React from 'react';
 import clsx from 'clsx';
 import Button, { ButtonProps } from '@material-ui/core/Button';
-import { MuiStyles, WithStyles, withStyles } from '@material-ui/core/styles';
+import {
+  experimentalStyled,
+  unstable_useThemeProps as useThemeProps,
+} from '@material-ui/core/styles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import { TypographyProps } from '@material-ui/core/Typography';
 import PickersToolbarText from './PickersToolbarText';
 import { ExtendMui } from './typings/helpers';
+import pickersToolbarButtonClasses, {
+  getPickersToolbarButtonUtilityClass,
+} from './pickersToolbarButtonClasses';
 
 export interface ToolbarButtonProps extends ExtendMui<ButtonProps, 'value' | 'variant'> {
   align?: TypographyProps['align'];
+  classes?: typeof pickersToolbarButtonClasses;
   selected: boolean;
   typographyClassName?: string;
   value: React.ReactNode;
   variant: TypographyProps['variant'];
 }
 
-export type PickersToolbarButtonClassKey = 'root';
+export type PickersToolbarButtonClassKey = keyof typeof pickersToolbarButtonClasses;
 
-export const styles: MuiStyles<PickersToolbarButtonClassKey> = {
-  root: {
-    padding: 0,
-    minWidth: '16px',
-    textTransform: 'none',
-  },
-};
+const useUtilityClasses = (styleProps: ToolbarButtonProps) => {
+  const { classes } = styleProps;
 
-const ToolbarButton: React.FunctionComponent<ToolbarButtonProps & WithStyles<typeof styles>> = (
-  props,
-) => {
-  const {
-    align,
-    classes,
-    className,
-    selected,
-    typographyClassName,
-    value,
-    variant,
-    ...other
-  } = props;
-
-  return (
-    <Button
-      data-mui-test="toolbar-button"
-      variant="text"
-      className={clsx(classes.root, className)}
-      {...other}
-    >
-      <PickersToolbarText
-        align={align}
-        className={typographyClassName}
-        variant={variant}
-        value={value}
-        selected={selected}
-      />
-    </Button>
-  );
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getPickersToolbarButtonUtilityClass, classes);
 };
 
-export default withStyles(styles, { name: 'MuiPickersToolbarButton' })(ToolbarButton);
+const PickersToolbarButtonRoot = experimentalStyled(
+  // TODO fix ts error
+  // @ts-expect-error
```

Comment:
How does this affect the return type?

Reply:
I got this.

![image](https://user-images.githubusercontent.com/18292247/116417058-23c1d180-a865-11eb-8ae8-024d44abe672.png)

Reply:
What is the type of `PickersToolbarButton` i.e. what type `experimentalStyled` create when called with an invalid type?

Reply:
fixed, the root cause it the `styles` type in overridesResolver(line: 43). So no change is needed for `experimentalStyled`